### PR TITLE
[timeseries] Implement recurrent forecasting model based on TabularPredictor and MLForecast

### PR DESCRIPTION
**Pull request by shchur**
*Description of changes:*
- Add a new model `RecursiveTabularModel` that uses the `TabularPredictor` to forecast time series 1 step into the future. The overall forecast is obtained by unrolling the model predictions over time.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: timeseries/src/autogluon/timeseries/models/presets.py [hunk 1]
```diff
@@ -38,6 +39,7 @@
     AutoETS=AutoETSModel,
     AutoARIMA=AutoARIMAModel,
     DynamicOptimizedTheta=DynamicOptimizedThetaModel,
+    RecursiveTabular=RecursiveTabularModel,
```
Comment (tonyhoo) [inline timeseries/src/autogluon/timeseries/models/presets.py:42]: which presets are we intended to add for this model?

File: timeseries/src/autogluon/timeseries/models/autogluon_tabular/mlforecast.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+
+    def predict(
+        self,
+        data: TimeSeriesDataFrame,
+        known_covariates: TimeSeriesDataFrame = None,
+        **kwargs,
+    ) -> TimeSeriesDataFrame:
+        predictions = self._predict_without_quantiles(data, known_covariates)
+        for q in self.quantile_levels:
+            predictions[str(q)] = predictions["mean"] + self.quantile_adjustments[q]
```
Comment (tonyhoo) [inline timeseries/src/autogluon/timeseries/models/autogluon_tabular/mlforecast.py:297]: Wondering how the quantile adjustment is done

File: timeseries/src/autogluon/timeseries/models/presets.py (hunk 1 above)
Comment (shchur) [reply timeseries/src/autogluon/timeseries/models/presets.py:42]: I plan to update these in a follow-up PR.

File: timeseries/src/autogluon/timeseries/models/autogluon_tabular/mlforecast.py (hunk 2 above)
Comment (shchur) [reply timeseries/src/autogluon/timeseries/models/autogluon_tabular/mlforecast.py:297]: Here we just use the empirical distribution of the quantiles to generate dummy quantiles. I will add another tree-based model that is trained to predict the quantiles using the wQL loss in a future PR.

**Review by tonyhoo — APPROVED** [review]

**Pull request merged**